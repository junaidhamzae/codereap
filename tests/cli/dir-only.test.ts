import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

describe('CLI --dirOnly directory aggregation', () => {
  it('aggregates directories and applies orphan rule', () => {
    const fixture = path.resolve(__dirname, '../fixtures/dir-only');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const res = runCli(['--root', fixture, '--out', outBase, '--dirOnly', '--onlyOrphans']);
    expect(res.status).toBe(0);
    const rows = res.json as any[];
    // Only orphan directories should be listed
    const dirs = rows.map(r => r.directory).sort();
    // b has an incoming edge from a, so it is not orphan; a has outgoing only, so it is orphan when onlyOrphans
    expect(dirs).toEqual(['a']);
    expect(rows).toMatchSnapshot();
  });
});


