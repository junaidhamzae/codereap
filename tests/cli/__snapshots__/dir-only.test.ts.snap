// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI --dirOnly directory aggregation aggregates directories and applies orphan rule 1`] = `
[
  {
    "directory": "a",
    "external-in-degree": 0,
    "file-count": 1,
    "orphan": true,
  },
]
`;
